linera-views/src/reentrant_collection_view.rs

```diff
@@ -372,15 +390,18 @@ where
     /// ```
     pub async fn contains_key(&self, short_key: &[u8]) -> Result<bool, ViewError> {
         let updates = self.updates.lock().await;
-        Ok(match updates.get(short_key) {
-            Some(entry) => match entry {
+        Ok(if let Some(entry) = updates.get(short_key) {
+            match entry {
                 Update::Set(_view) => true,
                 Update::Removed => false,
-            },
-            None => {
-                let key_index = self.context.base_tag_index(KeyTag::Index as u8, short_key);
-                !self.delete_storage_first && self.context.contains_key(&key_index).await?
             }
+        } else if self.delete_storage_first {
+            false
+        } else if self.cached_entries.lock().await.contains_key(short_key) {
+            true
+        } else {
+            let key_index = self.context.base_tag_index(KeyTag::Index as u8, short_key);
+            !self.delete_storage_first && self.context.contains_key(&key_index).await?
```

Comment:
we know that `!self.delete_storage_first` is true

Reply:
Thanks for catching this!